pkg/controller/hibernation/hibernation_controller.go

```diff
@@ -231,7 +231,11 @@ func (r *hibernationReconciler) Reconcile(ctx context.Context, request reconcile
 	clusterSync := &hiveintv1alpha1.ClusterSync{}
 	if err := r.Get(context.Background(), types.NamespacedName{Namespace: cd.Namespace, Name: cd.Name}, clusterSync); err != nil {
 		// This may be NotFound, which means the clustersync controller hasn't created the ClusterSync yet.
-		// Fail and requeue to wait for it to exist.
+		// It does so shortly after the cluster is marked as installed. Requeue immediately to "poll" for it.
```

Comment:
:heavy_check_mark: Error should be thrown by ClusterSync controller (it creates/attempts to create and requeue) instead of this one.